contracts/factory/factory.rs

```diff
@@ -13,14 +13,27 @@ pub struct ProxyFactory {
 #[near]
 impl ProxyFactory {
     #[init]
-    pub fn new(owner_id: AccountId) -> Self {
+    pub fn new() -> Self {
         assert!(!env::state_exists(), "Already initialized");
-        Self { owner_id }
+        Self {
+            owner_id: env::predecessor_account_id(), // TODO perhaps this can be removed completely since sub accounts can only be created by the factory owner
+        }
+    }
+
+    //TODO update - this should return the deposit if the txn fails
+    #[payable]
+    pub fn deposit_and_create_proxy(&mut self, owner_id: AccountId) -> Promise {
+        let deposit = env::attached_deposit();
```

Comment:
Would the factory owner be the one receiving this 3.8 near initially or is it the factory itself?

Reply:
NVM, I've seen that it's present on the transactions initiated on the frontend